src/hera/expr/_node.py

```diff
@@ -0,0 +1,224 @@
+from typing import Any, Dict, List, Optional, Union
```

Comment:
Any chance you are open to adding more inline documentation/docstrings to the elements that are implemented in this module? For long term maintenance I think it would be very helpful to people who will read this code. I think, primarily, class docstrings would be most helpful as independent methods, such as `__add__`, _should_ be understandable at first sight given they are part of the Python ecosystem; plus, the implementation is short and only gives new semantic meaning to the operators, which should be fairly accessible

Reply:
@flaviuvadan added some comments and docstrings to explain what is going on. Hopefully that helps.